### DISCUSSION TITLE
How to set the expect timeout for the playwright engine ?

### DISCUSSION THREAD

**Discussion by HugoImaios** [artilleryio/artillery#2728] [Artillery]
In playwright there is, along with timeout and navigation timeout, the expect timeout (https://playwright.dev/docs/test-timeouts#expect-timeout).

I've looked in artillery for a way to increase it instead of the default 5s of Playwright, but can't find it.

In playwright it can be set in the config file (so the equivalent in Artillery would be the yml config, maybe next to `defaultNavigationTimeout` ?). I think it can also be set on a per-test basis, via `test.use`, but anyway we don't have this on artillery (we only have `test.step` available).

If you have any clue to setting this expect timeout, let me know !

Have a nice day

**Comment by bernardobridge** [comment]
Hey @HugoImaios 👋 

You should be able to do something like:

```typescript
import { expect as baseExpect } from '@playwright/test';

const configuredExpect = baseExpect.configure({
  timeout: 10,
});
```

And then just use `configuredExpect` instead in your tests. This is based off a change another user contributed to Playwright, so it should work, but haven't tried it out myself. Let me know if that works for you!

**Reply by HugoImaios** [reply]
Oh great ! Thanks again for the quick reply @bernardobridge !
I'll try this et let you know !

**Reply by HugoImaios** [reply]
~~the downside is that it has to be in the playwright script, they can't be agnostics of the artillery-specific timeouts~~

**Reply by HugoImaios** [reply]
(and just to confirm to you : it works :slightly_smiling_face:  )

**Reply by bernardobridge** [reply]
> the downside is that it has to be in the playwright script, they can't be agnostics of the artillery-specific timeouts

Yes, we may look to improve that experience in the future, but right now as we don't use the Playwright Test Runner (and that's a test runner feature), this is the best workaround we are aware of 👍

**Reply by HugoImaios** [reply]
yeah totally !

**Reply by bernardobridge** [reply]
Btw @HugoImaios you could consider maybe controlling all these timeouts with environment variables instead (both in the config and the playwright test), then you could somehow make them depend on each other.

**Reply by HugoImaios** [reply]
I've though about it, but I've already got some env var for : the target env (one or two for headers), two csv files path, plus the multiple --config overrides. My devops coworkers are gonna be mad if I had another one :smile: 

But I've found another trick :wink: ! As the playwright config is a ts file, I've imported it to configure the expect.timeout at one place only :
```typescript
import { expect as baseExpect } from "@playwright/test";
import playwrightConfig from "../../playwright.config";

export const configuredExpect = baseExpect.configure({
    timeout: playwrightConfig.expect.timeout, // set it for both playwright and artillery
});
```
like this, I only set it in the playwright config (maybe reading the env vars some day, we'll see) :slightly_smiling_face:

**Reply by bernardobridge** [reply]
Oh that's a very interesting pattern @HugoImaios 🎉 , hadn't thought of that before :). Thanks for sharing - might steal that sometime for an example 🥷

**Reply by HugoImaios** [reply]
with pleasure ! I lack some time to participate as much as I would like, but if I can help :heart:

**Reply by ajhenry** [reply]
This works awesome, thanks a ton for this!
